inbm/dispatcher-agent/tests/unit/test_update_logger.py

```diff
@@ -194,3 +196,27 @@ def test_save_granular_log_file_sota_with_package_list(self, mock_run) -> None:
 
         assert mock_run.call_count == 2
         self.assertEqual(granular_log, expected_content)
+
+    @patch('dispatcher.update_logger.detect_os', return_value='TiberOS')
+    def test_save_granular_log_file_sota_with_tiberos_and_log(self, mock_os) -> None:
+        self.update_logger.ota_type = "sota"
+        self.update_logger._time = datetime.datetime(2024, 7, 3, 1, 50, 55, 935223)
+        log = {
+            "StatusDetail.Status": "Failed",
+            "FailureReason": "DownloadFail"
+        }
+        expected_content = {
+                            "UpdateLog": [
+                                    {
+                                        "StatusDetail.Status": "Failed",
+                                        "FailureReason": "DownloadFail"
+                                    }
+                                ]
+                            }
+
+        self.update_logger.save_granular_log_file(log)
+
+        with open(GRANULAR_LOG_FILE, 'r') as f:
```

Comment:
interacting with fs in unit test

Reply:
Updated. Use mock_open  for the file operation and verify the parameter passed in json.dump.